src/libraries/Microsoft.PowerFx.LanguageServerProtocol/JsonRpc/ObsoleteTypeSerializer/FormulaTypeJsonConverter.cs

```diff
@@ -1,86 +1,87 @@
-        {
-            var schemaPoco = FormulaTypeToSchemaConverter.Convert(value);
-            JsonSerializer.Serialize(writer, schemaPoco, _options);
-        }
-
-        private FormulaType GetFormulaType(FormulaTypeSchema schema)
-        {
-            return schema.Type switch
-            {
-                FormulaTypeSchema.ParamType.Number => FormulaType.Number,
-                FormulaTypeSchema.ParamType.String => FormulaType.String,
-                FormulaTypeSchema.ParamType.Boolean => FormulaType.Boolean,
-                FormulaTypeSchema.ParamType.Date => FormulaType.Date,
-                FormulaTypeSchema.ParamType.Time => FormulaType.Time,
-                FormulaTypeSchema.ParamType.DateTime => FormulaType.DateTime,
-                FormulaTypeSchema.ParamType.Color => FormulaType.Color,
-                FormulaTypeSchema.ParamType.Guid => FormulaType.Guid,
-                FormulaTypeSchema.ParamType.DateTimeNoTimeZone => FormulaType.DateTimeNoTimeZone,
-                FormulaTypeSchema.ParamType.Blank => FormulaType.Blank,
-                FormulaTypeSchema.ParamType.Hyperlink => FormulaType.Hyperlink,
-                FormulaTypeSchema.ParamType.UntypedObject => FormulaType.UntypedObject,
-                FormulaTypeSchema.ParamType.Record => GetAggregateType(schema.Fields, isTable: false),
-                FormulaTypeSchema.ParamType.Table => GetAggregateType(schema.Fields, isTable: true),
-
-                FormulaTypeSchema.ParamType.OptionSetValue => throw new NotImplementedException(),
-                FormulaTypeSchema.ParamType.EntityRecord => throw new NotSupportedException(),
-                FormulaTypeSchema.ParamType.EntityTable => throw new NotSupportedException(),
-                _ => throw new NotImplementedException(),
-            };
-        }
-
-        private FormulaType GetAggregateType(Dictionary<string, FormulaTypeSchema> fields, bool isTable)
-        {
-            Contracts.AssertValue(fields);
-            Contracts.AssertAllValues(fields);
-
-            var fieldsType = RecordType.Empty();
-            foreach (var field in fields)
-            {
-                if (fieldsType.FieldNames.Contains(field.Key))
-                {
-                    throw new NotSupportedException($"Multiple definitions of {field.Key}");
-                }
-
-                fieldsType = fieldsType.Add(field.Key, GetFormulaType(field.Value));
-            }
-
-            return isTable ? fieldsType.ToTable() : fieldsType;
-        }
-    }
-}
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol
+{
+    [Obsolete("Kept around short term to sequence changes in the formula bar")]
```

Comment:
>Obsolete [](http://example.com/codeflow?start=5&length=8)

Can the Obsolete message telll us what they should use instead?  (other clients may have started using this)(

Reply:
We may need to keep this public too (Obsolete is a chance for consumers to have a window to move onto the new stuff)

Reply:
There isn't really a legitimate use of this on the public surface, and part of the benefit of preview is that we can make changes like this. At the moment, Obsolete is primarily useful for us internally when hitting diamond dependency issues. I don't really see a reason to keep it `public` unless we know of a use case for it other than the formula bar.